src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -514,32 +514,32 @@ public int hashCode() {
 
     @ForceInline
     public static AbstractMemorySegmentImpl ofBuffer(Buffer b) {
-        // Implicit null check via NIO_ACCESS.scaleFactor(b)
-        final int scaleFactor = NIO_ACCESS.scaleFactor(b);
-        return ofBuffer(b, offset(b, scaleFactor), length(b, scaleFactor));
+        // Implicit null check via NIO_ACCESS.scaleShifts(b)
+        final int scaleShifts = NIO_ACCESS.scaleShifts(b);
+        return ofBuffer(b, offset(b, scaleShifts), length(b, scaleShifts));
     }
 
     @ForceInline
     private static AbstractMemorySegmentImpl ofBuffer(Buffer b, long offset, long length) {
         final Object base = NIO_ACCESS.getBufferBase(b);
         return (base == null)
-                ? arrayLessSegment(b, offset, length)
+                ? nativeSegment(b, offset, length)
```

Comment:
why not calling the other "heapSegment" ?

Reply:
Because heap segments don’t have to be backed by an array.

Reply:
> Because heap segments don’t have to be backed by an array.

I don't really know how to read this. We're asking a nio buffer to give us back a memory segment. There's two kinds of memory segment: heap segments and native segments (as the javadoc of memory segment clearly states). So, if `nativeSegment` is creating either a native segment (which can also be mapped), what does the other do, if not creating an heap segment?

Reply:
The reason was what @ExE-Boss mentioned but for me, I am happy with either name. I will update to `heapSegment()`.

Reply:
The method has been moved into Buffer implementations now.